PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1824,36 +1829,66 @@ void Notepad_plus::filePrint(bool showDialog)
 
 int Notepad_plus::doSaveOrNot(const TCHAR *fn)
 {
-	TCHAR pattern[64] = TEXT("Save file \"%s\" ?");
+	/*TCHAR pattern[64] = TEXT("Save file \"%s\" ?");
 	TCHAR phrase[512];
 	wsprintf(phrase, pattern, fn);
-	return doActionOrNot(TEXT("Save"), phrase, MB_YESNOCANCEL | MB_ICONQUESTION | MB_APPLMODAL);
+	return doActionOrNot(TEXT("Save"), phrase, MB_YESNOCANCEL | MB_ICONQUESTION | MB_APPLMODAL);*/
+
+	const TCHAR *params[1] = { fn };
+	return _nativeLangSpeaker.messageBox("NppSave", _pPublicInterface->getHSelf(),
+		TEXT("Save file \"$0$\" ?"),
+		TEXT("Save"), MB_YESNOCANCEL | MB_ICONQUESTION | MB_APPLMODAL, 1, params);
 }
 
 int Notepad_plus::doReloadOrNot(const TCHAR *fn, bool dirty)
 {
-	TCHAR* pattern = TEXT("%s\r\rThis file has been modified by another program.\rDo you want to reload it%s?");
+	/*TCHAR* pattern = TEXT("%s\r\rThis file has been modified by another program.\rDo you want to reload it%s?");
 	TCHAR* lose_info_str = dirty ? TEXT(" and lose the changes made in Notepad++") : TEXT("");
 	TCHAR phrase[512];
 	wsprintf(phrase, pattern, fn, lose_info_str);
 	int icon = dirty ? MB_ICONEXCLAMATION : MB_ICONQUESTION;
-	return doActionOrNot(TEXT("Reload"), phrase, MB_YESNO | MB_APPLMODAL | icon);
+	return doActionOrNot(TEXT("Reload"), phrase, MB_YESNO | MB_APPLMODAL | icon);*/
+
+	const TCHAR *params[1] = { fn };
+
+	if (dirty)
+	{
+		return _nativeLangSpeaker.messageBox("NppReloadDirty", _pPublicInterface->getHSelf(),
+			TEXT("$0$\r\rThis file has been modified by another program.\rDo you want to reload it and lose the changes made in Notepad++?"),
+			TEXT("Reload"), MB_YESNO | MB_ICONEXCLAMATION | MB_APPLMODAL, 1, params);
+	}
+	else
+	{
+		return _nativeLangSpeaker.messageBox("NppReload", _pPublicInterface->getHSelf(),
+			TEXT("$0$\r\rThis file has been modified by another program.\rDo you want to reload it?"),
+			TEXT("Reload"), MB_YESNO | MB_ICONQUESTION | MB_APPLMODAL, 1, params);
+	}
 }
 
 int Notepad_plus::doCloseOrNot(const TCHAR *fn)
 {
-	TCHAR pattern[128] = TEXT("The file \"%s\" doesn't exist anymore.\rKeep this file in editor?");
+	/*TCHAR pattern[128] = TEXT("The file \"%s\" doesn't exist anymore.\rKeep this file in editor?");
 	TCHAR phrase[512];
 	wsprintf(phrase, pattern, fn);
-	return doActionOrNot(TEXT("Keep non existing file"), phrase, MB_YESNO | MB_ICONQUESTION | MB_APPLMODAL);
+	return doActionOrNot(TEXT("Keep non existing file"), phrase, MB_YESNO | MB_ICONQUESTION | MB_APPLMODAL);*/
+
+	const TCHAR *params[1] = { fn };
+	return _nativeLangSpeaker.messageBox("NppKeep", _pPublicInterface->getHSelf(),
+		TEXT("The file \"$0$\" doesn't exist anymore.\rKeep this file in editor?"),
+		TEXT("Keep non existing file"), MB_YESNO | MB_ICONQUESTION | MB_APPLMODAL, 1, params);

```

Comment:
Same here:
No place holder `$INT_REPLACE$` neither `$STR_REPLACE$` in the message.
`$0$` is not the place holder.

Reply:
Please see this function https://github.com/notepad-plus-plus/notepad-plus-plus/pull/4111/files#diff-d70e145744162bf785ab5d44b36ad418R1464